pkg/reconciler/layer2.go

```diff
@@ -0,0 +1,157 @@
+package reconciler
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	networkv1alpha1 "github.com/telekom/das-schiff-network-operator/api/v1alpha1"
+	"github.com/telekom/das-schiff-network-operator/pkg/nl"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *reconcile) fetchLayer2() ([]networkv1alpha1.Layer2NetworkConfiguration, error) {
+	layer2List := &networkv1alpha1.Layer2NetworkConfigurationList{}
+	err := r.client.List(r.Context, layer2List)
+	if err != nil {
+		r.Logger.Error(err, "error getting list of Layer2s from Kubernetes")
+		return nil, err
+	}
+
+	nodeName := os.Getenv("HOSTNAME")
+	node := &corev1.Node{}
+	err = r.client.Get(r.Context, types.NamespacedName{Name: nodeName}, node)
+	if err != nil {
+		r.Logger.Error(err, "error getting local node name")
+		return nil, err
+	}
+
+	l2vnis := []networkv1alpha1.Layer2NetworkConfiguration{}
+
+	for _, item := range layer2List.Items {
+		if item.Spec.NodeSelector != nil {
+			selector, err := metav1.LabelSelectorAsSelector(item.Spec.NodeSelector)
+			if err != nil {
+				r.Logger.Error(err, "error converting nodeSelector of layer2 to selector", "layer2", item.ObjectMeta.Name)
+				return nil, err
+			}
+			if !selector.Matches(labels.Set(node.ObjectMeta.Labels)) {
+				r.Logger.Info("local node does not match nodeSelector of layer2", "layer2", item.ObjectMeta.Name, "node", nodeName)
+				return nil, err
+			}
+		}
+
+		l2vnis = append(l2vnis, item)
+	}
+
+	return l2vnis, nil
+}
+
+func (r *reconcile) reconcileLayer2(l2vnis []networkv1alpha1.Layer2NetworkConfiguration) error {
+	layer2Info := []nl.Layer2Information{}
```

Comment:
maybe call this `desired`, then it's a little clearer further down when comparing `existing` vs `desired`